apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/TypeTests.cs

```diff
@@ -494,6 +494,87 @@ private void AssertJsonEqual<T>(string expected, string actual) where T: IMessag
             });
         }
 
+        // Note: Value.Parse fails for list values containing null, hence the separate tests
+        [Fact]
+        public void SerializeStringArrayContainingNull()
+        {
+            var input = new[] { "x", null, "y" };
+            var expected = new Value { ListValue = new ListValue { Values = { Value.ForString("x"), Value.ForNull(), Value.ForString("y") } } };
+            var options = SpannerConversionOptions.Default;
+            Assert.Equal(expected, SpannerDbType.ArrayOf(SpannerDbType.String).ToProtobufValue(input, options));
+        }
+
+        [Fact]
+        public void DeserializeStringArrayContainingNull()
+        {
+            var protobuf = new Value { ListValue = new ListValue { Values = { Value.ForString("x"), Value.ForNull(), Value.ForString("y") } } };
+            var options = SpannerConversionOptions.Default;
+            var expected = new[] { "x", null, "y" };
+            Assert.Equal(expected, SpannerDbType.ArrayOf(SpannerDbType.String).ConvertToClrType<string[]>(protobuf, options));
+        }
+
+        [Fact]
+        public void SerializeNullableDoubleArrayContainingNull()
+        {
+            // The null value is converted into a 0
```

Comment:
This doesn't look right.

Reply:
Indeed, that one's wrong. Removing. The same comment on line 541 is correct though.